packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -336,14 +336,40 @@ export default function useOnBlockDrop(
 
 	const moveBlocks = useCallback(
 		( sourceClientIds, sourceRootClientId, insertIndex ) => {
+			const sourceBlocks = getBlocksByClientId( sourceClientIds );
+
+			// TODO: Figure out a better place for this. useOnBlockDrop shouldn't care about layout styles.
+			const unpinBlock = () => {
+				updateBlockAttributes(
```

Comment:
Tiny nit: would it be worth skipping calling `updateBlockAttributes` if no blocks need to be unpinned? I.e. move the `sourceBlocks.reduce` to be before this call?

I suppose this note is similar to your comment about `useOnBlockDrop shouldn't care about layout styles`, but unfortunately I can't think of a better place for this logic to live, either, without awkward `useEffect`s 🤔

Reply:
`updateBlockAttributes` is _supposed_ to do nothing if you pass it `{}`. (It doesn't, hence that change to the reducer that I have in this PR...)

Probably better to skip the call altogether though, yeah 👍